src/ac_training_lab/cobot280pi/server.py

```diff
@@ -0,0 +1,178 @@
+
+
+# MQTT Functions
+def on_connect(client, userdata, flags, rc, properties=None):
+    logger.info("Connection received with code %s." % rc)
+
+def on_publish(client, userdata, mid, properties=None):
+    logger.info("Successful publish.")
+
+def handle_message(msg, cobot):
+    # Parse payload to json dict
+    try:
+        payload_dict = json.loads(msg.payload)
+    except Exception as e:
+        return {
+            "success": False,
+            "error": str(e)
+        }
+
+    if "command" not in payload_dict:
+        return {
+            "success": False,
+            "error": "'command' key should be in payload"
+        }
+
+    # Match to a command function
+    cmd = payload_dict["command"]
+    if cmd == "control/angles":
+        return handle_control_angles(payload_dict["args"], cobot)
+    elif cmd == "control/coords":
+        return handle_control_coords(payload_dict["args"], cobot)
+    elif cmd == "control/gripper":
+        return handle_control_gripper(payload_dict["args"], cobot)
+    elif cmd == "query/angles":
+        return handle_query_angles(payload_dict["args"], cobot)
+    elif cmd == "query/coords":
+        return handle_query_coords(payload_dict["args"], cobot)
+    elif cmd == "query/gripper":
+        return handle_query_gripper(payload_dict["args"], cobot)
+    elif cmd == "query/camera":
+        return handle_query_camera(payload_dict["args"])
+    else:
+        return {
+            "success": False,
+            "error": "invalid command"
+        }
+
+if __name__ == "__main__":
+    logger = setup_logger()
+    task_queue = Queue()
+    args = parser.parse_args()
+
+    try:
+        cobot = MyCobot("/dev/ttyAMA0", 1000000)
+        logger.info("Cobot object initialized...")
+    except Exception as e:
+        logger.critical(f"could not initialize cobot with error {str(e)}")
+        if not args.debug:
+            logger.info("exiting...")
+            sys.exit(1)
+        else:
+            cobot = None
+
+    def on_message(client, userdata, msg):
+        logger.info(
+            f"Recieved message with: \n\ttopic: {msg.topic}\n\tqos: {msg.qos}\n\tpayload: {msg.payload}")
+        task_queue.put(msg)
+
+    client = paho.Client(client_id="", userdata=None, protocol=paho.MQTTv5)
+    client.on_connect = on_connect
+    client.on_message = on_message
+    client.on_publish = on_publish
+
+    client.tls_set(tls_version=paho.ssl.PROTOCOL_TLS)
+    client.username_pw_set(HIVEMQ_USERNAME, HIVEMQ_PASSWORD)
+    client.connect(HIVEMQ_HOST, DEVICE_PORT)
+    client.subscribe(DEVICE_ENDPOINT, qos=2)
+    client.loop_start()
+    logger.info("Ready for tasks...")
+
+    while True:
+        msg = task_queue.get()  # blocks if queue is empty
+        response_dict = handle_message(msg, cobot)
+        client.publish(
+            RESPONSE_ENDPOINT,
+            qos=2,
+            payload=json.dumps(response_dict)
+        )
+        time.sleep(3)
```

Comment:
Consider removing the sleep (since there's a blocking line already) and using a wait_for_publish instead.